### PR TITLE
support basic authenticate

### PR DESCRIPTION
**Pull request by wuwx**

### CODE REVIEW

File: src/inet/http_server.c [hunk 1]
```diff
@@ -138,7 +145,7 @@ static int read_request(struct http_server_t *self_p,
                         struct http_server_request_t *request_p)
 {
     int res;
-    char buf[128];
```
Comment (eerimoq) [inline src/inet/http_server.c]: What's the reason for increasing the buffer size? As it is an embedded system we must always try to minimize the memory usage.

Comment (wuwx) [inline src/inet/http_server.c]: in some browser, user-agent header is very big.
some user use the browser visit device http server
and then device will bei crashed.....

may be we can ignore user-agent header, alse a better idear

Comment (eerimoq) [inline src/inet/http_server.c]: Yeah let's fix the logic in ``read_header_line()`` so it does not write outside the buffer. Read into the buffer, and when it is full just ignore the rest of the data.

Comment (wuwx) [inline src/inet/http_server.c]: how about 401 response ?

**Comment by wuwx** [comment]
how about 401 response ?

**Comment by wuwx** [comment]
the test hangs on 

```
enter: test_request_auth
0:debug:http_conn_0:default: GET /auth.html HTTP/1.1
0:debug:http_conn_0:default: Authorization: Basic YWRtaW46YWRtaW4=
0:debug:http_conn_0:default: User-Agent: TestcaseRequestIndex
0:debug:http_conn_0:default: Connection: keep-alive
0:debug:http_conn_0:default: Connection thread 'http_conn_0' waiting for a new connection.
```

i will find out

**Comment by wuwx** [comment]
aha~~, test code it's ok now

**Comment by eerimoq** [comment]
:P

**Pull request merged**